tools/ir-generator/ir-generator.ypp

```diff
@@ -142,6 +142,16 @@ element
     : irclass           { global.push_back($1); }
     | COMMENTBLOCK      { global.push_back(new CommentBlock(@1, $1)); }
     | COMMENTBLOCK irclass {
+            /* find all the comments that have been pushed on the global stack */
+            while (global.size() > 0) {
+                auto c = dynamic_cast<CommentBlock *>(global.back());
+                if (c != nullptr) {
+                    $2->comments.push_back(c);
+                    global.pop_back();
```

Comment:
This reverses the order of the comments.  If you want to pull in all the comments immediately before the class, get rid of the `COMMENTBLOCK irclass` rule and scan over the trailing comments to pull them into the class in the bare `irclass` rule.  Alternately, add a rule to consecutive comments into a single comment:

    comment_block : COMMENTBLOCK { $$ = new CommentBlock(@1, $1); }
        | comment_block COMMENTBLOCK {
            $$ = $1;
            $$->srcInfo += @2;
            $$->body += $2; }